### PR TITLE
Bump rstar and robust dependencies

### PR DESCRIPTION
**Pull request by urschrei**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/main/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

**Comment by urschrei** [comment]
Ah yes, need to bump CI containers first of course.

**Comment by urschrei** [comment]
bors retry

**Comment by urschrei** [comment]
bors cancel

**Comment by urschrei** [comment]
This ended up being slightly more invasive than I thought it would be as I had to add an `use-rstar-11` feature and corresponding impls too. If people would prefer a more comprehensive `CHANGES` entry lmk

**Review by michaelkirk — APPROVED** [review]

File: geo/CHANGES.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 * Implement "Closest Point" from a `Point` on a `Geometry` using spherical geometry. <https://github.com/georust/geo/pull/958>
 * Bump CI containers to use libproj 9.2.1
+* Bump rstar and robust dependencies
```
[the remark this answers is not in the document]
Comment (michaelkirk) [reply geo/CHANGES.md:7]: (breaking for geo, non-breaking for geo-types)

[the remark this answers is not in the document]
Comment (michaelkirk) [reply geo/CHANGES.md:7]: Also, can you add a note to the geo-types CHANGES as well?

**Comment by urschrei** [comment]
bors r=michaelkirk

**Pull request merged**